text/0000-try-expr.md

```diff
@@ -0,0 +1,916 @@
+
+Whatever keyword is chosen, it can't be contextual.
+
+As with `catch { .. }`, the syntactic form `<word> { .. }` where `<word>`
+is replaced with any possible keyword would conflict with a struct named
+`<word>` as seen in this perfectly legal snippet in Rust 2015,
+where `<word>` has been substituted for `try`:
+
+```rust
+struct try;
+fn main() {
+    try {
+    };
+}
+```
+
+### Aside note:
+
+The snippet above emits the following warning:
+
+```
+warning: type `try` should have a camel case name such as `Try`
+```
+
+which is also the case for `catch`.
+This warning decreases the risk that someone has defined a type named `try`
+anywhere in the ecosystem which happens to be beneficial to us.
+
+## For reserving `try` specifically
+
+This is discussed in the [rationale for `try`][rationale for try].
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+The keyword `try` will be reserved.
+This will allow you to write expressions such as:
+
+```rust
+try {
+    let x = foo?;
+    let y = bar?;
+    // Note: OK-wrapping is assumed here, but it is not the goal of this RFC
+    // to decide either in favor or against OK-wrapping.
+    x + y
+}
+```
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+[list of keywords]: https://doc.rust-lang.org/book/second-edition/appendix-01-keywords.html#keywords-currently-in-use
+
+The word `try` is reserved as a keyword in the [list of keywords]
+in Rust edition 2018 and later editions.
+
+The keyword `try` is used in "try expressions" of the form `try { .. }`.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+There are two main drawbacks to the `try` keyword.
+
+## Association with exception handling - Both a pro and con
+
+> I think that there is a belief – one that I have shared from time to time – that it is not helpful to use familiar keywords unless the semantics are a perfect match, the concern being that they will setup an intuition that will lead people astray. I think that is a danger, but it works both ways: those intuitions also help people to understand, particularly in the early days. So it’s a question of “how far along will you get before the differences start to matter” and “how damaging is it if you misunderstand for a while”.
+>
+> [..]
+>
+> Rust has a lot of concepts to learn. If we are going to succeed, it’s essential that people can learn them a bit at a time, and that we not throw everything at you at once. I think we should always be on the lookout for places where we can build on intuitions from other languages; it doesn’t have to be a 100% match to be useful.
+
+\- [Niko Matsakis](https://internals.rust-lang.org/t/bikeshed-rename-catch-blocks-to-fallible-blocks/7121/4)
+
+For some people, the association to `try { .. } catch { .. }` in languages such
+as Java, and others in the [prior-art] section, is unhelpful wrt. teachability
+because they see the explicit, reifed, and manually propagated exceptions in
```

Comment:
`reifed` or [`reified`](https://github.com/rust-lang/rfcs/pull/2388/files#diff-909563f501605136130c7eb468511337R111)?

Reply:
The latter :) Thanks for catching this typo!